### ISSUE TITLE
Migrate Kudos details to use assets directly vs dynamic avatar endpoint

### ISSUE THREAD

**Issue by mbeacom** [gitcoinco/web#3129]
#### Story

As a user, I want avatar images on the kudos details page to be loaded from the kudos owner's `Profile.avatar_url` to benefit from caching.

#### User Interaction

The user requests a kudos details page, for example:  https://gitcoin.co/kudos/588/ether_gun

#### Current Behavior

Avatar images are served from: `https://gitcoin.co/dynamic/avatar/@<github_handle>`

#### Expected Behavior

Avatar images served from:  `https://c.gitcoin.co/avatars/<salt>/<github_handle>.svg` which can be accessed via  `Avatar.get_avatar_url()` or  the `avatar_url` template tag.

**Comment by kuhnchris** [comment]
![image](https://user-images.githubusercontent.com/761911/49687471-61dac100-fb03-11e8-9477-4e923427fe36.png)
:suspect:

**Comment by kuhnchris** [comment]
@mbeacom could this be a general problem with avatar_url? Same happens on the profile page
![image](https://user-images.githubusercontent.com/761911/49696376-7ec9cf80-fba9-11e8-83db-ae571f45152a.png)

**Comment by kuhnchris** [comment]
Just checked, the avatar/templatetags/avatar_tags.py basically does this:



@register.simple_tag
def avatar_url(profile, use_svg=True):
    if isinstance(profile, Profile) and profile.avatar:
        return profile.avatar.get_avatar_url(use_svg)
    if isinstance(profile, str):
        return f'{settings.BASE_URL}dynamic/avatar/{profile}'
    return f'{settings.BASE_URL}dynamic/avatar/Self'

But since we always provide a string in the templates, wouldn't that ALWAYS result in the dynamic path?

**Comment by mbeacom** [comment]
Right.  We don't pass a string to this in *every* place.  I had to include that to accommodate the mounds of places we're using a view to serve an image...  It's been like this for quite a while.  I migrated some places, but not all since a good amount of refactor is necessary.

We *should* be using `Avatar.avatar_url` which defaults to the user media assets and only falls back to the "dynamic" (gh avatar or gh default, both of which are already served by GH...) if necessary.

The point of *this* issue is to migrate AWAY from using that view anywhere.  The string check is merely a fallback because it's happening all over, sadly.

**Comment by kuhnchris** [comment]
That's weird cause the original post of yours reads like you want us to use avatar_url.
So the point of this issue is that we use Avatar.avatar_url everywhere?
Wouldn't it be easier to simply add that to the avatar_url templatetag and have the tag itself fallback to the other one?

Just an idea here, the alternative is to grep for that template in all the files and replace it with the corresponding logic...

**Comment by mbeacom** [comment]
`avatar_url` template tag **already** uses `Avatar.avatar_url`

The whole point of this github issue is:  

```
Current Behavior
Avatar images are served from: https://gitcoin.co/dynamic/avatar/@<github_handle>

Expected Behavior
Avatar images served from: https://c.gitcoin.co/avatars/<salt>/<github_handle>.svg
```

**Comment by kuhnchris** [comment]
OK, let me rephrase it, since either i'm deaf or we have some communication issues:
- Currently some views use avatar_url
- Some do not
- Those avatar_url (and Avatar.get_avatar_url()) point to the wrong destination (dynamic/avatar).
- Those occourences need to be replaced by Profile.avatar_url which should point to the S3

Do I get you right now?

**Comment by mbeacom** [comment]
It's not so much that they're pointing to the **wrong** place.  It's a matter of pointing to a view that requires an additional request being placed against **our** web app versus cloudfront -> s3 bucket.

We shouldn't be proxying an image if we could simply serve it from CF/S3 over the separate uri: `c.gitcoin.co`

The point here is to eliminate the additional requests against `https://gitcoin.co/dynamic/avatar/@<github_handle>` in favor of: `https://c.gitcoin.co/avatars/<salt>/<github_handle>.svg` as specified in the original spec.

**Comment by kuhnchris** [comment]
ok, we still seem to talk past each other - ill try once more else someone else has to take this:
I am aware that we currently serve from "our" server (gitcoin.co) by using the /dynamic/ path.
Your goal is that we serve it from S3/CF/whever, which is where c.gitcoin.co points to. 
This URL, along with the salt, is getting generated (or stored) in Profile.avatar_url right?
Hence tl;dr of the task:
- [ ] Make sure all occurrences of ```gitcoin.co/dynamic/avatar/...``` point at ```c.gitcoin.co/avatars/<salt>/<github_handle>.svg``` by replacing ```avatar_url``` and ```Avatar.get_avatar_url()```

**Comment by mbeacom** [comment]
There is no need to replace `Avatar.get_avatar_url()` - This method **handles** the above case.  We need to update all of the frontend/django templates to use the avatar media asset urls (provided by `avatar_url` if `Profile` is passed versus a string).

So, the django template needs to be updated to use MTV versus making numerous requests against a proxy view.  Realistically, we should simply be using `avatar_url` templatetag, but by passing the actual objects versus strings...  This allows us to **slowly** transition **all** of the views over to use the `Avatar.svg|github_svg` without removing the fallback.
